src/backend/processor/metadata.ts

```diff
@@ -0,0 +1,189 @@
+    }
+
+    /** Reloads metadata after updating. */
+    private updateMetadataPath() {
+        this.disposeWatcher();
+
+        this.metadataPath = workspace.getConfiguration('codechecker.backend').get<string>('outputFolder');
+
+        if (this.metadataPath) {
+            const workspaceFolder = workspace.workspaceFolders![0].uri.fsPath;
+
+            // Substitute basic variables into the folder path
+            this.metadataPath = this.metadataPath
+                .replace(/\${workspaceRoot}/g, workspaceFolder)
+                .replace(/\${workspaceFolder}/g, workspaceFolder)
+                .replace(/\${cwd}/g, process.cwd())
+                .replace(/\${env\.([^}]+)}/g, (sub: string, envName: string) => process.env[envName] ?? '');
+
+            this.metadataPath = Uri.joinPath(Uri.file(this.metadataPath), './metadata.json').fsPath;
+
+            // Create watcher that triggers on updates
+            this.metadataWatch = workspace.createFileSystemWatcher(this.metadataPath);
+
+            this.metadataWatchDisposables.push(this.metadataWatch.onDidChange(this.reloadMetadata, this));
+            this.metadataWatchDisposables.push(this.metadataWatch.onDidCreate(this.reloadMetadata, this));
+            this.metadataWatchDisposables.push(this.metadataWatch.onDidDelete(this.reloadMetadata, this));
+        }
+
+        this.reloadMetadata()
+            .catch((err) => {
+                console.log(err);
+                window.showErrorMessage('Unexpected error when reloading metadata\nCheck console for more details');
+            });
+    }
+
+    /** Reloads associated CodeChecker metadata.
+     *
+     * Automatically fires in the following cases:
+     *   - Extension config changed
+     *   - Contents of the metadata file changed
+     *   - ``codechecker.backend.reloadMetadata`` fired
+     */
+    async reloadMetadata(): Promise<void> {
+        let precheckFailed = false;
+
+        if (!this.metadataPath) {
+            window.showWarningMessage(
+                'Metadata folder has invalid path\n' +
+                'Please change `CodeChecker > Backend > Output folder path` in the settings'
+            );
+
+            precheckFailed = true;
+        }
+
+        if (!workspace.workspaceFolders?.length) {
+            window.showInformationMessage('CodeChecker is disabled - open a workspace to get started');
+
+            precheckFailed = true;
+        }
+
+        if (precheckFailed) {
+            this._metadata = undefined;
+            this._metadataSourceFiles = new Map();
+            this._metadataUpdated.fire(this._metadata);
+            return;
+        }
+
+        let metadata;
+
+        try {
+            metadata = await parseMetadata(this.metadataPath!);
+        } catch (err) {
+            switch (err.code) {
+            // Silently ignore File not found errors
+            case 'FileNotFound':
+            case 'EntryNotFound':
+                break;
```

Comment:
I started the plugin and I tried to reload the metadata by the command and with a metadata.json file which content is an empty json object (`{}`) and the plugin didn't give me any error. Also if the metadata file doesn't exist the plugin also not give me any message so I guessed that the metadata is reloaded but it's not true. It would be good to give proper error messages to the user similar to this one:

```suggestion
            case 'FileNotFound':
                window.showErrorMessage('metadata.json file does not exist!');
                break;

            case 'EntryNotFound':
                window.showErrorMessage('Invalid metadata.json file format!');
                break;
```

Reply:
On FileNotFound there's a reason to not show any notifications.

An user can load any workspace, including non-C++ and non-Codechecker repositories, and in unrelated workspaces the extension should be disabled.
Since there will never be a CodeChecker file or folder in this case, this is a quick way to check for an unrelated workspace.

What I can do is show an info-level message, with the exact same message that's in the default sidebar.
When there's a better on-off switch (maybe a toolbar button?) this check should be changed.

---

I cannot reproduce the ``{}``-issue, it threw the expected `SyntaxError: required member not found` for me.
(Apparently EntryNotFound doesn't exist, cleaned that up.)

Reply:
I explicitly wanted to skip the 'FileNotFound' type errors, because in a non-C++ or non-CodeChecker repository, the notification would always show up.
The FileNotFound error is a quick way to filter out unrelated repos.
And even if there's no notification, a quick-start message is displayed on the CodeChecker tab.

When there's a more convenient way to disable CodeChecker (maybe a button in the toolbar?) this will show a notification.

---

I couldn't reproduce the `{}` issue, it throws the expected ``SyntaxError: required member not found``, which is handled properly.
(Apparently ``EntryNotFound`` doesn't exist, fixed that as well.)